hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/HoodieFlinkCopyOnWriteTable.java

```diff
@@ -153,6 +154,27 @@ public HoodieWriteMetadata<List<WriteStatus>> delete(
     return new FlinkDeleteCommitActionExecutor<>(context, writeHandle, config, this, instantTime, keys).execute();
   }
 
+  /**
+   * Delete the given prepared records from the Hoodie table, at the supplied instantTime.
+   *
+   * <p>This implementation requires that the input records are already tagged, and de-duped if needed.
+   *
+   * <p>Specifies the write handle explicitly in order to have fine-grained control with
+   * the underneath file.
+   *
+   * @param context {@link HoodieEngineContext}
+   * @param instantTime Instant Time for the action
+   * @param preppedRecords Hoodie records to delete
+   * @return {@link HoodieWriteMetadata}
+   */
+  public HoodieWriteMetadata<List<WriteStatus>> deletePrepped(
```

Comment:
why we have 2 deletePrepped methods?

Reply:
This also applies to other operations such as upsert. One set of methods are called from `HoodieFlinkWriteClient` by casting the object to `HoodieFlinkTable`. The other set of methods are derived from `HoodieTable`.